src/Mumble.proto

```diff
@@ -20,6 +20,23 @@ message Version {
 	optional string os_version = 4;
 }
 
+// Sent after the version packet, indicate the protocol preference of the client.
+// - This message is not required for text-only clients. It can be sent before or after
+//   the Authenticate message. However, it has to be sent before transmitting any audio data.
+// - The server will reply with a Capabilities message as well, with one item in the list,
+//   indicating the protocol to be used.
+// - In the middle of a session, a client can perform this negotiation again by sending another
+//   valid Capabilities message. The server may also request the client to use another protocol
+//   by sending a Capabilities message with the protocol it'd like to switch to.
+message Capabilities {
```

Comment:
> The description is tailored to a specific voice protocol negotiation, but the name `Capabilities` seems to imply a more general use-case.

IIRC the original idea was to create a general message type, where people can add additional feature flags into this message. I also recalled I wrote in another issue that the codec information is dispersed among many message types (but in the end I think people felt we should mark them as obsolete because Opus will be the only option).

The description might seem to be a little bit narrow. I can redo the description. But do we really want to add more items to this message in the future? If not, I can consider renaming it into something like `ProtocolCapabilities`.


> And in fact, I think this all makes sense to be bundled together with the encryption methods the client knows. Thus, I think the description should be generalized a bit.

I don't quite follow this... I think you felt the name `Capabilities` is over-generalized...

Reply:
> But do we really want to add more items to this message in the future? If not, I can consider renaming it into something like ProtocolCapabilities.

I would think so, yes. So in general I was thinking that this could really contain flags for all the different features the Mumble protocol supports in principle and then clients and servers can specify which of these they do in fact support.